src/js/angular/clustermanagement/directives/cluster-list.directive.js

```diff
@@ -0,0 +1,198 @@
+import {Location} from "../../models/clustermanagement/cluster";
+
+const modules = [];
+angular.module('graphdb.framework.clustermanagement.directives.cluster-list', modules)
+    .directive('clusterList', ClusterListComponent);
+
+ClusterListComponent.$inject = ['$translate', '$timeout', '$repositories', 'productInfo', 'toastr', 'RemoteLocationsService', 'ClusterContextService'];
+
+function ClusterListComponent($translate, $timeout, $repositories, productInfo, toastr, RemoteLocationsService, ClusterContextService) {
+    return {
+        restrict: 'E',
+        templateUrl: 'js/angular/clustermanagement/templates/cluster-list.html',
+        link: ($scope, element, attrs) => {
+            // =========================
+            // Private variables
+            // =========================
+            const subscriptions = [];
+
+            // =========================
+            // Public variables
+            // =========================
+            $scope.rowHeights = {};
+            $scope.editedNodeIndex = undefined;
+            $scope.allSuggestions = [];
+            $scope.loader = false;
+            $scope.cluster = undefined;
+            $scope.clusterView = undefined;
+            $scope.errors = [];
+            $scope.newLocation = '';
+
+            /**
+             * Adds a new (empty) node to the list of cluster nodes for editing.
+             */
+            $scope.addNode = () => {
+                $scope.clusterView.push({});
+                $scope.editedNodeIndex = $scope.clusterView.length - 1;
+            };
+
+            /**
+             * Filters suggestions based on user input for node's endpoint.
+             * @param {Object} node - The node object with an endpoint property.
+             */
+            $scope.filterSuggestions = (node) => {
+                const query = node.endpoint && node.endpoint.toLowerCase();
+                if (query) {
+                    $scope.suggestions = $scope.allSuggestions.filter((item) => item.toLowerCase().includes(query));
+                    $scope.showDropdown = true;
+                } else {
+                    $scope.showDropdown = false;
```

Comment:
If the endpoint is available, we calculate `$scope.suggestions.` However, if it is not, we might need to reset `$scope.suggestions` to its default state.

Reply:
fixed